### PR TITLE
[DOC] Document the release process

### LINKED ISSUE #17: [DOC] Document the release process
At least
- how to update the bpmn-visualization js lib
- git tag
- update the version in DESCRIPTION
- CRAN publishing: #10 
- R package release good practices

Tasks
- [x] Decide where to store this documentation: in the **contributing guide**
- [x] write the documentation #41

### PR DESCRIPTION
**Pull request by csouchet**
Closes #17

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -100,6 +100,10 @@ git push --force-with-lease 25-customize_overlays
 
 At this point, you're ready to make your changes! Feel free to ask for help. Everyone is a beginner at first :smile_cat:
 
+## BPMN Visualization update
```
Comment (tbouffard) [inline CONTRIBUTING.md:103]: Clarify this is the js lib update
```suggestion
## bpmn-visualization-js update
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -100,6 +100,10 @@ git push --force-with-lease 25-customize_overlays
...
+
+- Change [bpmn-visualization.min.js](inst/htmlwidgets/lib/bpmn-visualization/bpmn-visualization.min.js)
```
Comment (tbouffard) [inline CONTRIBUTING.md:105]: Explain how to retrieve the js file
```suggestion
- Change [bpmn-visualization.min.js](inst/htmlwidgets/lib/bpmn-visualization/bpmn-visualization.min.js). The file can be retrieved from [jsdelivr](https://www.jsdelivr.com/package/npm/bpmn-visualization) or [unpkg](https://unpkg.com/browse/bpmn-visualization/).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -111,3 +115,63 @@ A PR can only be merged into master by a maintainer, if all of these conditions
...
+
+Create a new PR and merge it, if the version in the [DESCRIPTION](./DESCRIPTION) file is not good.
+
+#### GitHub issues and milestones
+
+**Note:** we always put issues related to a version in a Milestone whose name matches the version.
+
+- Ensure the name of the milestone used for the new release version matches the name of the tag/version that has just been pushed. Renamed it if needed.
+- Clean this opened milestone if some issues are still opened (move them to a new one or discard milestone from them)
+- Close the milestone
```
Comment (tbouffard) [inline CONTRIBUTING.md:133]: ⚠️ we are not using this way of doing in this repository, so I suggest we remove this paragraph

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -111,3 +115,63 @@ A PR can only be merged into master by a maintainer, if all of these conditions
...
+  This ensures that development can continue without impacting the writing of the content of the in progress release. That way,
+  if a PR is merged, `release-drafter` will update the `Next` draft release keeping the in-progress release untouched.
+- The draft release for the newly tagged version should already exist:
+   - [release-drafter](https://github.com/release-drafter/release-drafter) creates or updates draft release for the
+     next version each time a pull request is merged to the `master` branch.
+   - create a new release if it is missing or rename the existing one to match .
+- Assign the new tag as release target and save the draft (this should have already been managed by `release-drafter`)
+- Ensure that `This is a pre-release` is unchecked (except if we are releasing alpha, beta, rc, ...)
+- In the release description (check previous releases as a source of inspiration)
+   - at least add/update a link to the related milestone
```
Comment (tbouffard) [inline CONTRIBUTING.md:148]: ⚠️ not releavant

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -111,3 +115,63 @@ A PR can only be merged into master by a maintainer, if all of these conditions
...
+
+#### Twitter
+
+You can use this template:
+
+> 📣 BPMN Visualization - R package {version} is out! 🎉
+>
+> ===> some short description here <===
+>
+> #bpmnvisualization #bpmn #visualization #R #opensource
```
Comment (tbouffard) [inline CONTRIBUTING.md:174]: :question: use `#R` or `#RLang` (https://twitter.com/search?q=%23Rlang&src=typed_query) or both

I have also seen #Rlang`, #rlang`

Comment (csouchet) [reply CONTRIBUTING.md]: Twitter doesn't differentiate `#Rlang` and `#rlang` 😉

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -111,3 +115,60 @@ A PR can only be merged into master by a maintainer, if all of these conditions
...
+- Verify the tag version as the same as the release note.
+- Published the release **only when you are done** with the release content. 
+
+#### Publish the new version on CRAN
+
+- Pull the new version created by the release workflow: `git pull origin`
+- Ensure you are on the last tag locally: `git checkout vX.Y.Z`
+- Open the project in the R Studio
+- On the **Build** tab, click on **More** > **Build Binary Package**
+<p align="center"><img title="Build Binary Package" src="./doc/r-studio-binary-build.png" alt="Build Binary Package"></p>
```
Comment (tbouffard) [inline CONTRIBUTING.md:154]: ⚠️ we have to decide where to store the image used by the internal documentation. In #45, I put the readme image in a `img` folder, but we should only have one folder.
I don't know if `doc` is ok for R project, especially when we will implement #40. But for now, I am ok to use the `doc` folder.

@csouchet do we keep the `doc` folder? If so, I will update the other PR.

Comment (csouchet) [reply CONTRIBUTING.md:154]: I don't have preferences. I just reproduce what we did on bpmn-visualization-js.

Comment (tbouffard) [reply CONTRIBUTING.md:154]: ok, let's start with the `doc` folder

Comment (tbouffard) [reply CONTRIBUTING.md:154]: Integrated in #45 which has been merged in the `main` branch

**Review by tbouffard — APPROVED** [review]

**Pull request merged**